src/chains/ethereum/ethereum/src/wallet.ts

```diff
@@ -636,7 +638,40 @@ export default class Wallet {
       fakePrivateKey = Buffer.allocUnsafe(32).fill(0, 0, 31);
       fakePrivateKey[31] = 1;
     } else {
-      fakePrivateKey = Buffer.concat([addressBuf, addressBuf.slice(0, 12)]);
+      // Private keys must not be greater than *or equal to*:
+      // 0xfffffffffffffffffffffffffffffffebaaedce6af48a03bbfd25e8cd0364141, and
+      // if they are that large then signing of the transaction (ecdsaSign) will
+      // fail.
+      // Historically, we've just concatenated the address with part of itself,
+      // to make something long enough to use as a private key, so we can't
+      // change the default/legacy behavior now. But for addresses that would
+      // generate an invalid private key we must use a different approach. If
+      // the legacy way of generating a private key results in a key that is too
+      // large
+      //
+      const first12 = addressBuf.slice(0, 12);
+      fakePrivateKey = Buffer.concat([addressBuf, first12]);
+      // BigInt is slow, so only do it if we might need to, which we usually
+      // never will.
+      // Since we already have a slice of the first 12 bytes let's use it to
+      // help check if we might overflow the max secp256k1 key value. If the
+      // first 12 bytes, the most significant digits of the key, are all `255`
+      // then there is a chance that the fakePrivateKey will be too large.
+      // `Buffer.compare` is a native method in V8, so it should be pretty fast.
+      // example: the address `0xffffffffffff{anything can go here}` generates a
+      // bad fakePrivateKey but, `0xfffffffffffe{anything can go here}` never
+      // will. There are obviously many chances for a false positive, but the
+      // next condition in the `while` loop will catch those.
+      if (first12.compare(TWELVE_255s) === 0) {
+        // keccak returns a 32 byte hash of the input data, which is the exact
+        // length we need for a private key.
+        // note: if keccak can return its own input as its output, then this
+        // loops forever. The chances of this happening are impossibly low, so
+        // it's not worth the effort to check, but it would be interesting if
+        // someone reported an issue that can cause this for a specific
+        // address!
+        fakePrivateKey = keccak(fakePrivateKey);
```

Comment:
```suggestion
        while (
          BigInt(`0x${fakePrivateKey.toString("hex")}`) >=
          SECP256K1_MAX_PRIVATE_KEY
        ) {
          // keccak returns a 32 byte hash of the input data, which is the exact
          // length we need for a private key.
          // note: if keccak can return its own input as its output, then this
          // loops forever. The chances of this happening are impossibly low, so
          // it's not worth the effort to check, but it would be interesting if
          // someone reported an issue that can cause this for a specific
          // address!
          fakePrivateKey = keccak(fakePrivateKey);
        }
```